chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -67,6 +90,7 @@ sealed abstract class Bits(dirArg: Direction, width: Width, override val litArg:
     if (isLit()) {
       Bool(((litValue() >> x.toInt) & 1) == 1)
     } else {
+      Binding.checkSynthesizable(this, s"'this' ($this)")
```

Comment:
The main implication of this (somewhat) duplicated code is that changes in FIRRTL require any corresponding checking code in chisel3 frontend to be updated, which is a manual (and therefore error-prone) effort.

Alternative (not necessarily better, though up for debate) solutions may be to have chisel3 check reference validity at time of FIRRTL generation (or perhaps even earlier, at the time the reference is made, so a stack trace is meaningful)? And shouldn't source locators make the undefined references debuggable too? - it should tell you which line of Chisel source generated the bad line of code